### PR TITLE
Am user login

### LINKED ISSUE #10: Iteration 2: Create Layout for Login Page
Create a login page with inputs for username, password & submit button that allows the user to login

### LINKED ISSUE #11: Iteration 2: Login Page Post Request
Upon submitting the correct username & password, a post request is made to obtain the user's credentials

API Endpoint:  https://rancid-tomatillos.herokuapp.com/api/v2/login

### LINKED ISSUE #12: Iteration 2: Login Page Functionality
- [ ] After logging in: the user will know that they are logged in

- [ ] they will be taken back to the homepage

- [ ] they will have the ability to log out

### LINKED ISSUE #13: Iteration 2: User Logged In Functionality
- [ ] After logging in: the user will know that they are logged in

- [ ] they will be taken back to the homepage

- [ ] they will have the ability to log out

### LINKED ISSUE #14: Iteration 2: User Log Out Functionality
When a user clicks the logout button, they are taken back to the homepage & the app shows that no one is logged in

### PR DESCRIPTION
**Pull request by aemcdonald**
## Description
Created Login component & imported files where appropriate
Created method in Form to render login form & display on DOM
Created methods in Form to obtain user input from login form & set state
Added route path to take user to main page upon successful login
Created POST request to post user login 
Created method in App to call POST method & update state in App
Created method in Login to receive props from App & set state in Login to logged in user's info
Added conditional rendering functionality to display login/logout in App
Created method in App to update state when a user logs out

## Motivation and Context
Added an issue for a bug, extra login button persists through user login & logged in user being routed back to main page. Button does not impede application functionality in any way.
closes #10 
closes #11 
closes #12 
closes #13 
closes #14 

## How Has This Been Tested?
Tested in browser for appropriate functionality
Will write robust test suite on a separate branch

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.

### CODE REVIEW

File: src/ApiCalls.js [hunk 1]
```diff
@@ -4,6 +4,19 @@ const ApiCalls = {
...
+    return fetch('https://rancid-tomatillos.herokuapp.com/api/v2/login', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(userLoginInfo)
+    })
+    .then(response => response.json())
+    .then(data => data)
+    .catch(err => console.log('failed', err.message))
```
Comment (Kristiannmiller) [inline src/ApiCalls.js:19]: We should probably be doing more with this error message than just console.log. Would love to display something to the user if the information doesn't match what's in the database.

File: src/App/App.js [hunk 2]
```diff
@@ -1,28 +1,73 @@
...
 
-
 class App extends Component {
   constructor() {
     super();
     this.state = {
-      user: {id: 1, name: "Olivia", email: "alan@turing.io"}
+      user: {},
+      isOnHomePage: true
+      //we will want these to be empty strings eventually...
```
Comment (Kristiannmiller) [inline src/App/App.js:16]: Let's get rid of some of these vestigial comments on the next PR

File: src/App/App.js [hunk 3]
```diff
@@ -1,28 +1,73 @@
...
+    }
+  }
+
+  updateCurrentUser = (userInfo) => {
+    console.log("App pre state", this.state)
+    this.setState(userInfo)
+    console.log("App state", this.state)
+  }
+
+  handleLoginSubmit = (userInfo) => {
```
Comment (Kristiannmiller) [inline src/App/App.js:26]: I think we can get rid of the extra console.log here now

File: src/App/App.js [hunk 4]
```diff
@@ -1,28 +1,73 @@
...
     this.state = {
-      user: {id: 1, name: "Olivia", email: "alan@turing.io"}
+      user: {},
+      isOnHomePage: true
+      //we will want these to be empty strings eventually...
+    }
+  }
+
+  updateCurrentUser = (userInfo) => {
+    console.log("App pre state", this.state)
```
Comment (Kristiannmiller) [inline src/App/App.js:21]: I think we can get rid of the extra console.log here now

File: src/App/App.js [hunk 5]
```diff
@@ -1,28 +1,73 @@
...
+    this.setState(userInfo)
+    console.log("App state", this.state)
+  }
+
+  handleLoginSubmit = (userInfo) => {
+    ApiCalls.postUserLogin(userInfo)
+    .then(data => {console.log(data)
```
Comment (Kristiannmiller) [inline src/App/App.js:28]: I think we can get rid of the extra console.log here now

File: src/Login/Login.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+      password: ''
+      //name: 'Olivia', email: 'olivia@turing.io', password: 'pword'
+    }
+  }
+
+  getUserInput = event => {
+    this.setState({ [event.target.name]: event.target.value })
+  }
+
+  submitLogin = event => {
```
Comment (Kristiannmiller) [inline src/Login/Login.js:21]: This was a tricky one to figure out! Glad we discovered how to pass a function to a child component. Looks good!

**Comment by Kristiannmiller** [comment]
This looks great! We've still got that bug with the original login button still showing up on the login page. Let's revisit that bug later and move on to some testing and minor css work with the next steps.

File: src/App/App.js (hunk 4 above)
Comment (letakeane) [reply src/App/App.js:21]: Don't forget to be using your react dev tools :D They'll show you a more accurate, up to date version of your state.

File: src/App/App.js [hunk 7]
```diff
@@ -1,28 +1,73 @@
...
-      user: {id: 1, name: "Olivia", email: "alan@turing.io"}
+      user: {},
+      isOnHomePage: true
+      //we will want these to be empty strings eventually...
+    }
+  }
+
+  updateCurrentUser = (userInfo) => {
+    console.log("App pre state", this.state)
+    this.setState(userInfo)
```
Comment (letakeane) [inline src/App/App.js:22]: Don't forget that setState expects to be given an object that specifies which part of state is being updated. By passing it the userInfo object, you may lose other parts of state you don't intend to overwrite, such as isOnHomepage

File: src/Login/Login.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    event.preventDefault();
+    const loginInfo = {
```
Comment (letakeane) [inline src/Login/Login.js:23]: You can actually just put state in the handleSubmit! `this.props.handleSubmit(this.state)`

**Pull request merged**